base/string_utils.cpp

```diff
@@ -259,6 +259,21 @@ bool EndsWith(std::string const & s1, std::string const & s2)
   return s1.size() >= s2.size() && s1.compare(s1.size() - s2.size(), s2.size(), s2) == 0;
 }
 
+bool EndsWith(strings::UniString const & s1, std::string const & s2)
```

Comment:
The first 128 unicode code points are valid ascii so the function is reasonable as long as s1 is an ascii string.